openeogeotrellis/job_tracker_v2.py

```diff
@@ -85,48 +87,112 @@ class YarnAppReportParseException(Exception):
+            )
+        return auth
+
     def get_job_metadata(self, job_id: str, user_id: str, app_id: str) -> _JobMetadata:
-        try:
-            command = ["yarn", "application", "-status", app_id]
-            with TimingLogger(f"Running {command}", logger=_log.debug):
-                application_report = subprocess.check_output(
-                    command, stderr=subprocess.PIPE
-                ).decode("utf-8")
-        except CalledProcessError as e:
-            stdout = e.stdout.decode()
-            if "doesn't exist in RM or Timeline Server" in stdout:
-                raise AppNotFound(stdout) from None
-            else:
-                raise
-        return self.parse_application_report(application_report)
+        # TODO: Are there any cases were we *do* need to provide credentials to the Kerberos authentication?
+        #   At present we are assuming there is already a Kerberos ticket because kinit was run before
+        #   the geopyspark driver process was started.
+        #   AFAIK that means we don't have to provide user & pw which is in fact the whole point.
+
+        url = self.get_application_url(application_id=app_id)
+        response = requests.get(url, auth=self.get_authentication_provider())
+
+        # Handle error responses: We need the field "message" from "RemoteException"
+        # See https://hadoop.apache.org/docs/current/hadoop-yarn/hadoop-yarn-site/WebServicesIntro.html
+        #   specifically, the section: Response Errors
+        #
+        # Example of an error responses body:
+        # {
+        #     "RemoteException": {
+        #         "exception": "NotFoundException",
+        #         "message": "java.lang.Exception: app with id: application_1674538064532_46723 not found",
+        #         "javaClassName": "org.apache.hadoop.yarn.webapp.NotFoundException"
+        #     }
+        # }
+        if response.status_code == 404:
+            remote_exc = response.json().get("RemoteException", {})
+            message = remote_exc.get(
+                "message",
+                "Unknown error. Error response did not contain the message property.",
+            )
+            raise AppNotFound(message) from None
+        else:
+            # Check if there was any other HTTP error status.
+            response.raise_for_status()
+            return self.parse_application_response(json=response.json())
+
+    @staticmethod
+    def _ms_epoch_to_date(epoch_millis: int) -> Union[str, None]:
+        """Parse millisecond timestamp from app report and return as rfc3339 date (or None)"""
+        if epoch_millis == 0:
+            return None
+        utc_datetime = dt.datetime.utcfromtimestamp(epoch_millis / 1000)
+        return rfc3339.datetime(utc_datetime)
+
+    @classmethod
+    def parse_application_response(cls, json: dict) -> _JobMetadata:
+        if not json:
+            raise YarnAppReportParseException("Response is empty")
```

Comment:
Unnecessary check because `response.json()` will already have failed.

Reply:
Indeed. Removed it.